test/h.test.js

```diff
@@ -83,6 +83,7 @@ test("skip null and Boolean children", () => {
 
 test("components", () => {
   const Component = (data, children) => h("div", data, children)
+  const Wrapper = (data, children) => h("div", data, children.map(vn => h(Component, null, vn)))
```

Comment:
I've added this component and the corresponding tests to assert the type of the children argument. The implementation of h guarantees that children will always be any array and this test validates that. I discovered this because the typing of the `Component` interface was wrong.